### DISCUSSION TITLE
How should I approach the copyleft license of cypress-cloud?

### DISCUSSION THREAD

**Discussion by Dekis** [currents-dev/cypress-currents-reporter#145] [Q&A] [answered]
We've been using @currents/cli with Cypress which is under MIT license. But as of Cypress 12.6.0+ you'd need to use cypress-cloud moving forward. This however seems to be a mix of MIT(for the copied parts) and GPL-3.0-or-later. For copyleft my understanding is that you should then have the host software (so my application in this case) have the same license. This is not possible for me - does this mean there's no way of using currents with newer version of cypress for commercial projects anymore?

**Comment by agoldis** [comment, marked as answer]
@Dekis 

The intent was to ensure that forks and derivatives are available to the public. 

**You don't need to have the application under test to be GPL-licensed**

- you can use the tool internally with no problem
- you can copy and change the code internally without distributing it for public use
- if you create a derivative copy - e.g. create a new package that provides similar functionality or enhances the functionality and distribute (in a standalone form, or as part of) you need to have the same license, have authors attribution and reference to the origin (see the license wording)

Hope that helps, if you need to discuss your specific situation please connect with us privately support at currents dot dev
